exchanges/okx/helpers.go

```diff
@@ -0,0 +1,204 @@
+	"strings"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+)
+
+// orderTypeFromString returns order.Type instance from string
+func orderTypeFromString(orderType string) (order.Type, error) {
+	orderType = strings.ToLower(orderType)
+	switch orderType {
+	case orderMarket:
+		return order.Market, nil
+	case orderLimit:
+		return order.Limit, nil
+	case orderPostOnly:
+		return order.PostOnly, nil
+	case orderFOK:
+		return order.FillOrKill, nil
+	case orderIOC:
+		return order.ImmediateOrCancel, nil
+	case orderOptimalLimitIOC:
+		return order.OptimalLimitIOC, nil
+	case "mmp":
+		return order.MarketMakerProtection, nil
+	case "mmp_and_post_only":
+		return order.MarketMakerProtectionAndPostOnly, nil
+	case "twap":
+		return order.TWAP, nil
+	case "move_order_stop":
+		return order.TrailingStop, nil
+	case "chase":
+		return order.Chase, nil
+	default:
+		return order.UnknownType, fmt.Errorf("%w %v", order.ErrTypeIsInvalid, orderType)
+	}
+}
+
+// orderTypeString returns a string representation of order.Type instance
+func orderTypeString(orderType order.Type) (string, error) {
+	switch orderType {
+	case order.ImmediateOrCancel:
+		return "ioc", nil
+	case order.Market, order.Limit, order.Trigger,
+		order.PostOnly, order.FillOrKill, order.OptimalLimitIOC,
+		order.MarketMakerProtection, order.MarketMakerProtectionAndPostOnly,
+		order.Chase, order.TWAP, order.OCO:
+		return orderType.Lower(), nil
+	case order.ConditionalStop:
+		return "conditional", nil
+	case order.TrailingStop:
+		return "move_order_stop", nil
+	default:
+		return "", fmt.Errorf("%w %v", order.ErrUnsupportedOrderType, orderType)
+	}
+}
+
+// getAssetsFromInstrumentID parses an instrument ID and returns a list of assets types
+// that the instrument is associated with
+func (ok *Okx) getAssetsFromInstrumentID(instrumentID string) ([]asset.Item, error) {
+	if instrumentID == "" {
+		return nil, errMissingInstrumentID
+	}
+	pf, err := ok.CurrencyPairs.GetFormat(asset.Spot, true)
+	if err != nil {
+		return nil, err
+	}
+	splitSymbol := strings.Split(instrumentID, pf.Delimiter)
+	if len(splitSymbol) <= 1 {
+		return nil, fmt.Errorf("%w %v", currency.ErrCurrencyNotSupported, instrumentID)
+	}
+	pair, err := currency.NewPairDelimiter(instrumentID, pf.Delimiter)
+	if err != nil {
+		return nil, err
+	}
+	switch {
+	case len(splitSymbol) == 2:
+		resp := make([]asset.Item, 0, 2)
+		enabled, err := ok.IsPairEnabled(pair, asset.Spot)
+		if err != nil {
+			return nil, err
+		}
+		if enabled {
+			resp = append(resp, asset.Spot)
+		}
+		enabled, err = ok.IsPairEnabled(pair, asset.Margin)
+		if err != nil {
+			return nil, err
+		}
+		if enabled {
+			resp = append(resp, asset.Margin)
+		}
+		if len(resp) > 0 {
+			return resp, nil
+		}
+	case len(splitSymbol) > 2:
+		switch splitSymbol[len(splitSymbol)-1] {
```

Comment:
```suggestion
		switch strings.ToLower(splitSymbol[len(splitSymbol)-1]) {
```
and then you don't have to repeat yourself in the cases.